### ISSUE TITLE
Mypy is reporting incorrect line numbers on a 2.7kloc files.

### ISSUE THREAD

**Issue by fjovell-swish** [python/mypy#21902]
**Bug Report**

Trying to typecheck the OCPP package, and when running mypy against [v201/enums.py](https://github.com/mobilityhouse/ocpp/blob/b0e0b07fd77652a586b0b6aba1bac77b6683573a/ocpp/v201/enums.py) file, I get the following error:

```console
$ mypy ocpp/v201/enums.py
```

**Expected Behavior**

Line numbers to  be correctly displayed and errors to make sense in the context.

**Actual Behavior**

```
ocpp/v201/enums.py:1485: error: Incompatible types in assignment (expression has type "StandardizedVariableName", base class "str" defined the type as "Callable[[str, SupportsIndex | None, SupportsIndex | None], int]")  [assignment]
ocpp/v201/enums.py:2649: error: Incompatible types in assignment (expression has type "SpacesCountSignageVariableName", base class "str" defined the type as "Callable[[str, SupportsIndex | None, SupportsIndex | None], int]")  [assignment]
Found 2 errors in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: mypy 2.3.1 (compiled: yes)
- Mypy command-line flags: --strict
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: python3.11
- OS: MacOS Tahoe 26.6.2

**Comment by fjovell-swish** [comment]
Word adding: I tried with a warm cache and without cache (by deleting the entire cache) and I got the same result.

**Comment by JelleZijlstra** [comment]
Is the version you linked exactly the same as what you're type checking? Line 1487 of https://github.com/mobilityhouse/ocpp/blob/b0e0b07fd77652a586b0b6aba1bac77b6683573a/ocpp/v201/enums.py#L1487 correctly has that error (the `count` enum member unsafely overrides the `str.count` method).

**Comment by fjovell-swish** [comment]
@JelleZijlstra You are completely right! I totally missed this. Thank you for your comment.
